apps/freeablo/faworld/inventory.cpp

```diff
@@ -316,6 +316,34 @@ namespace FAWorld
         return total;
     }
 
+    void Inventory::takeOutGold(int32_t quantity)
+    {
+        for (auto row : boost::irange(mInventoryBox.height() - 1, -1, -1))
+            for (auto column : boost::irange(mInventoryBox.width() - 1, -1, -1))
+            {
+                auto& item = mInventoryBox.get(column, row);
+                if (item.getType() != ItemType::gold)
+                    continue;
+                auto toTake = std::min(quantity, item.mCount);
+                item.mCount -= toTake;
+                quantity -= toTake;
+                if (item.mCount == 0)
+                    item = {};
+                if (quantity == 0)
+                    return;
+            }
+        release_assert("Not enough gold");
```

Comment:
Side note, but this doesn't work as intended, as release_assert expects a boolean, it will cast the const char* to bool, and get true, so this will never fail. Can be worked aroudn by `release_assert(false && "message")`, but I've also added a  `message_and_abort` macro for thsi case. I also added `invalid_enum` for invalid enum values.

Reply:
Will consider in the future. Thanks!